src/components/Delegation/DelegationCenter.js

```diff
@@ -91,40 +126,39 @@ export default injectIntl(
     withStateHandlers(
       {
         busy: false,
+        showPoolWarning: false,
+        reputationInfo: {},
+        selectedPools: [],
       },
       {
         setBusy: () => (busy) => ({busy}),
+        setShowPoolWarning: () => (showPoolWarning) => ({showPoolWarning}),
+        setReputationInfo: () => (reputationInfo) => ({reputationInfo}),
+        setSelectedPools: () => (selectedPools) => ({selectedPools}),
       },
     ),
     withHandlers({
-      handleOnMessage: ({navigation, setBusy}) => async (event, intl) => {
-        setBusy(true)
+      navigateToDelegationConfirm: ({intl, navigation, setBusy}) => async (
+        selectedPools,
+      ) => {
         try {
-          const pools: Array<SelectedPool> = JSON.parse(
-            decodeURI(event.nativeEvent.data),
-          )
+          setBusy(true) // trigger "please wait" modal
```

Comment:
not specific to just here, but instead of doing setBusy with `finally`s everywhere which is error prone during refactoring, maybe it would have made more sense to make setBusy take an argument that's a lambda which guarantees setBusy is always set to false after the lambda is done running. Something like

```
wrapBusy(func: void => Promise<void>): Promise<void> {
    setBusy(true);
    try {
        await func();
    } finally {
        setBusy(false);
    }
}
```

Reply:
@SebastienGllmt I added a wrapper as you suggested. I see the point on doing this but we must be careful when using this pattern with a `navigation.navigate()` call. If after navigating the origin component is unmounted, then `setBusy` will try to update the State of an unmounted component (which in this case will cause the please wait modal getting stuck forever). In this case it's safe doing it because the origin component is kept in the stack.

I'm leaving this PR unmerged for a while in case you want to review the last changes.